hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestDatanodeReport.java

```diff
@@ -172,8 +172,20 @@ public void testDatanodeReportMissingBlock() throws Exception {
         // all bad datanodes
       }
       cluster.triggerHeartbeats(); // IBR delete ack
-      lb = fs.getClient().getLocatedBlocks(p.toString(), 0).get(0);
-      assertEquals(0, lb.getLocations().length);
+      int c = 0;
+      while (true) {
+        lb = fs.getClient().getLocatedBlocks(p.toString(), 0).get(0);
+        if (0 != lb.getLocations().length) {
+          c++;
+          if (c > 7) {
+            assertEquals(0, lb.getLocations().length);
```

Comment:
The `assertEquals` condition is contradictory to the condition `(0 != lb.getLocations().length)`.

If the intent is to fail the test here, just use `Assert.fail("getLocatedBlocks failed after 7 retries")` instead.